src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/security/AuthorisedCaseSearchOperation.java

```diff
@@ -53,36 +57,56 @@ public AuthorisedCaseSearchOperation(
-            .orElse(CaseSearchResult.EMPTY);
+    public CaseSearchResult execute(CrossCaseTypeSearchRequest searchRequest) {
+        List<CaseType> authorisedCaseTypes = getAuthorisedCaseTypes(searchRequest);
+        CrossCaseTypeSearchRequest authorisedSearchRequest = createAuthorisedSearchRequest(authorisedCaseTypes, searchRequest);
+
+        return searchCasesAndFilterFieldsByAccess(authorisedCaseTypes, authorisedSearchRequest);
     }
 
-    private CaseSearchResult searchCasesAndFilterFieldsByAccess(CaseType caseType, CaseSearchRequest caseSearchRequest) {
-        CaseSearchResult result = caseSearchOperation.execute(caseSearchRequest);
-        filterFieldsByAccess(caseType, result.getCases());
-        return result;
+    private List<CaseType> getAuthorisedCaseTypes(CrossCaseTypeSearchRequest searchRequest) {
+        return searchRequest.getCaseTypeIds()
+            .stream()
+            .map(caseTypeId -> authorisedCaseDefinitionDataService.getAuthorisedCaseType(caseTypeId, CAN_READ).orElse(null))
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
     }
 
-    private void filterFieldsByAccess(CaseType caseType, List<CaseDetails> cases) {
-        cases.forEach(caseDetails -> {
-            filterCaseFieldsByAclAccess(caseType, caseDetails);
-            filterCaseFieldsBySecurityClassification(caseDetails);
-        });
+    private CrossCaseTypeSearchRequest createAuthorisedSearchRequest(List<CaseType> authorisedCaseTypes, CrossCaseTypeSearchRequest originalSearchRequest) {
+        List<String> authorisedCaseTypeIds = authorisedCaseTypes.stream().map(CaseType::getId).collect(Collectors.toList());
+        return new CrossCaseTypeSearchRequest(authorisedCaseTypeIds, originalSearchRequest.getSearchRequestJsonNode());
     }
 
-    private void filterCaseFieldsByAclAccess(CaseType caseType, CaseDetails caseDetails) {
-        JsonNode data = objectMapperService.convertObjectToJsonNode(caseDetails.getData());
-        JsonNode filteredData = accessControlService.filterCaseFieldsByAccess(data, caseType.getCaseFields(), getUserRoles(), CAN_READ);
-        caseDetails.setData(objectMapperService.convertJsonNodeToMap(filteredData));
+    private CaseSearchResult searchCasesAndFilterFieldsByAccess(List<CaseType> authorisedCaseTypes, CrossCaseTypeSearchRequest authorisedSearchRequest) {
+        return Optional.of(authorisedCaseTypes)
+            .filter(CollectionUtils::isNotEmpty)
+            .map(caseTypes -> {
+                CaseSearchResult result = caseSearchOperation.execute(authorisedSearchRequest);
+                filterFieldsByAccess(caseTypes, result.getCases());
+                return result;
+            }).orElse(CaseSearchResult.EMPTY);
+    }
+
+    private void filterFieldsByAccess(List<CaseType> authorisedCaseTypes, List<CaseDetails> cases) {
+        cases.forEach(caseDetails -> authorisedCaseTypes.stream()
```

Comment:
mmm this might be a bit slow when the list of cases is big?
How about organising the List<CaseType> authorisedCaseTypes in a Map with key the case type?
Also, if you want to leave it as it is now, better to create the authorisedCaseTypes.stream() once and not once for each case

Reply:
yes map seems good, implemented.